### ISSUE TITLE
NEMO_THREADED=1 does not work

### ISSUE THREAD

**Issue by wbhart** [Nemocas/Nemo.jl#784]
Setting NEMO_THREADED=1 in the environment before starting Julia does not cause Nemo to allow threading.

The reason is that __isthreaded is set outside of __init__ and is therefore hard coded at the time of the last precompilation. It is not changed by simply changing the environment and starting Julia again.

In order to set the value when Nemo is started up, independently of precompilation, it must be set in __init__

This is probably the case for NEMO_SOURCE_BUILD and any other flags we rely on for various things.

**Comment by thofma** [comment]
I agree about the thread variable.

Regarding `NEMO_SOURCE_BUILD`. This is set in `deps/build.jl`, which is not part of the Nemo module and therefore not precompiled. So changes will always be picked up.

Are there any other environment variables/flags that we pick up? I could not find any.

**Comment by wbhart** [comment]
I didn't check if there are others. You are probably right. This probably only affects the one environment variable.
